### PR TITLE
Use default application credential

### PR DESCRIPTION
**Pull request by kroikie**
Update config quickstart to use default application credential which is best practice for server side Firebase authentication.

### CODE REVIEW

File: config/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Getting started
 
 1. [Add Firebase to your Android Project](https://firebase.google.com/docs/android/setup).
 2. Create a service account as described in [Adding Firebase to your Server](https://firebase.google.com/docs/admin/setup) and download the JSON file.
-  - Copy the private key JSON file to this folder and rename it to `service-account.json`.
+  - Set `GOOGLE_APPLICATION_CREDENTIALS` environment variable to path of downloaded credential file.
```
Comment (samtstern) [inline config/README.md]: Maybe show how you can use `gcloud auth login default` (or whatever it is) to get it working without even having to download the JSON file?

Comment (egilmorez) [reply config/README.md]: I'm curious about that too.  The best practice is to set that var if you're running locally or in a non-Google server environment, correct?  But this *would* work in GCP/GAE without the var properly set, right Arthur?

File: config/index.js [hunk 2]
```diff
@@ -10,24 +10,27 @@ var SCOPES = ['https://www.googleapis.com/auth/firebase.remoteconfig'];
 
 /**
  * Get a valid access token.
+ *
+ * This method assumes it is either running in a trusted Google environment like GCP or if running
```
Comment (egilmorez) [inline config/index.js]: Suggest disambiguating "it" with "your server" or whatever fits.

File: config/README.md (hunk 1 above)
Comment (samtstern) [reply config/README.md]: Even locally I never set the environment variable unless I am trying to run two different things with different service accounts.  Normally the default machine-wide credential Just Works :tm:

Comment (kroikie) [reply config/README.md]: The `gcloud auth application-default login` command does not work with some of the Firebase APIs so lets go with the env var for now and change later on when/if gcloud gets wider support.

File: config/index.js (hunk 2 above)
Comment (kroikie) [reply config/index.js]: @egilmorez not sure what you mean here could you give more detail?

Comment (kroikie) [reply config/index.js]: @egilmorez updated the docs here, could you have another look?

**Comment by kroikie** [comment]
Will only merge when documentation that uses these snippets are ready.

File: config/index.js [hunk 3]
```diff
@@ -9,24 +9,27 @@ var SCOPES = ['https://www.googleapis.com/auth/firebase.remoteconfig'];
...
-    var jwtClient = new google.auth.JWT(
-      key.client_email,
-      null,
-      key.private_key,
-      SCOPES,
-      null
-    );
-    jwtClient.authorize(function(err, tokens) {
-      if (err) {
+    google.auth.getApplicationDefault(function (err, authClient) {
```
Comment (hiranya911) [inline config/index.js]: https://github.com/googleapis/google-auth-library-nodejs is the recommended library here.

Comment (kroikie) [reply config/index.js]: Done!

**Comment by rvalitov** [comment]
I think using `GOOGLE_APPLICATION_CREDENTIALS` is not a universal approach. I prefer to keep JWT, because in this case I can use it anywhere, for example in Heroku, or other services where I may not have the rights to set environmental variables.

**Comment by jhuleatt** [comment]
Closing since this is outdated - confirmed with @kroikie over chat

**Pull request closed without merging**